packages/wallet/src/services/DRepRegistrationTracker.ts

```diff
@@ -0,0 +1,31 @@
+import { Cardano } from '@cardano-sdk/core';
+import { Observable, map } from 'rxjs';
+import { TrackerSubject } from '@cardano-sdk/util-rxjs';
+
+export interface CreateDRepRegistrationTrackerProps {
+  historyTransactions$: Observable<Cardano.HydratedTx[]>;
+}
+
+export const createDRepRegistrationTracker = ({
+  historyTransactions$
+}: CreateDRepRegistrationTrackerProps): TrackerSubject<boolean> =>
+  new TrackerSubject(
+    historyTransactions$.pipe(
+      map((txs) => {
+        const reverseTxs = txs.reverse();
+
+        for (const {
+          body: { certificates }
+        } of reverseTxs) {
+          if (certificates) {
+            for (const certificate of certificates) {
+              if (certificate.__typename === Cardano.CertificateType.UnregisterDelegateRepresentative) return false;
+              if (certificate.__typename === Cardano.CertificateType.RegisterDelegateRepresentative) return true;
```

Comment:
@mkazlauskas we left this out since the wallet won't let you sign a DRep certificate if it isn't your drep key. however this _could_ just be a limitation imposed by the SDK, so we might want to also add this check @vetalcore for enhanced protection

Reply:
@mkazlauskas would something like works?
```
dRepKeyHash = wallet.getPubDRepKey();
certificate.dRepCredential.hash === Crypto.Hash28ByteBase16.fromEd25519KeyHashHex(dRepKeyHash)
```
do i need to check for `dRepCredential.type` as well?

Reply:
Depends on what this observable means, as I mentioned [here](https://github.com/input-output-hk/cardano-js-sdk/pull/1192#discussion_r1547531912) I don't fully understand it. cc @mchappell

Reply:
It's simply there to track whether the wallet has registered as a delegate representative @mkazlauskas

Reply:
I suggest renaming to `isRegisteredAsDrep$` then. @vetalcore that hash check seems right.

Reply:
done

Reply:
Please squash the commits before merging and add a footer like this in the commit message:
```
BREAKING CHANGE: hoist ObservableWallet getPubDRepKey under ObservableWallet.delegation
```